Lun 732

zoliszeredi opened the pull request:
Non latin filenames in zip files raise UnicodeDecodeError when trying to extract.

Commit:
LUN-732: Zip decode unicode filenames.

Commit:
LUN-732: Switch to unicode error messages.

bhodorog commented on filer/models/archivemodels.py:
```diff
@@ -69,7 +71,8 @@ def _collisions_zip(self, filer_file):
         cwd_path = cwd.pretty_logical_path + u'/'
         no_end_slash = lambda x: x[:-1] if x.endswith('/') else x
-        zip_paths = [cwd_path + no_end_slash(x) for x in zippy.namelist()]
+        zip_paths = [cwd_path + decode_str(no_end_slash(x))
```
Can `decode_str(no_end_slash(x))`  be also `x.rstrip("/").decode('utf8')`. ?
I'm thinkin about same length, no supplementary lambda's needed and std libs.

bhodorog commented on filer/models/archivemodels.py:
```diff
@@ -84,15 +87,16 @@ def _extract_zip(self, filer_file):
+    def _create_parent_folders(self, filename):
         """Creates the folder parents for a given entry."""
-        dir_parents_of_entry = entry.filename.split(os.sep)[:-1]
+        dir_parents_of_entry = filename.split(os.sep)[:-1]
```
is the os.sep safe here? (assumes the code is ran on a OS with '/' as separator)
I notice in _collisions_zip method you're using '/'. If os.sep is safe here, I'm thinking you should replace '/' from above.

Commit:
LUN-732: Zip unicode support code review;

1. Replace '/' with os.sep
2. Remove decode lambda
3. Can extract unfilled archives.

Commit:
Remove ipdb import.
